Core/ELF/ParamSFO.cpp

```diff
@@ -276,3 +267,27 @@ void ParamSFOData::ValueData::SetData(const u8* data, int size)
 	u_size = size;
 }
 
+std::string ParamSFOData::GenerateFakeID(std::string filename) {
+	// Generates fake gameID for homebrew based on it's folder name.
+	// Should probably not be a part of ParamSFO, but it'll be called in same places.
+	std::string file = PSP_CoreParameter().fileToStart;
+	if (filename != "")
+		file = filename;
+
+	std::size_t lslash = file.find_last_of("/");
+	file = file.substr(lslash + 1);
+
+	int sumOfAllLetters = 0;
+	for (char &c : file) {
+		sumOfAllLetters += c;
+		c = toupper(c);
+	}
+
+	if (file.size() < 4) {
+		file += "HOME";
+	}
+	file = file.substr(0, 4);
+
+	std::string fakeID = file + StringFromFormat("%05i", sumOfAllLetters);
```

Comment:
Nitpicking: I'm trying to consistently use %d for integers in format strings, as %i is a little scary in sscanf and better to be consistent. There are bunch more of these but I'm not gonna comment on each one.